src/utils/url.ts

```diff
@@ -87,7 +87,45 @@ export function urlParams(url: string): Record<string, ParsedParam | ParsedParam
   return result
 }
 
+export function urlParamsArray(url: string): [string, ParsedParam | ParsedParam[]][] {
+  const params = _allParams(url)
+  const result: [string, ParsedParam | ParsedParam[]][] = []
+  Object.keys(params).forEach((k) => { result.push([k, _parseParam(params, k)]) })
+  return result
+}
+
 export function getQueryParameter(url: string, name: string): ParsedParam | ParsedParam[] {
   const params = _allParams(url)
   return _parseParam(params, name)
 }
+
+export class ParsedUrl {
+  hash: string
+  host: string
+  hostname: string
+  pathname: string
+  port: string
+  protocol: string
+  search: string
+
+  constructor (url: string) {
+    // eslint-disable-next-line
```

Comment:
I was getting an error about not needing to escape the `/`, but having 2 in a row makes it a comment 😅

Reply:
it should be a comment 👍